Alignment/OfflineValidation/plugins/DiMuonVertexValidation.cc

```diff
@@ -323,25 +345,40 @@ void DiMuonVertexValidation::analyze(const edm::Event& iEvent, const edm::EventS
   // fill the VtxProb plots
   VtxProbPlots.fillPlots(SVProb, tktk_p4);
 
+  math::XYZPoint MainVertex(0, 0, 0);
+  const reco::Vertex* theClosestVertex = nullptr;
   // get collection of reconstructed vertices from event
   edm::Handle<reco::VertexCollection> vertexHandle = iEvent.getHandle(vertexToken_);
-
-  math::XYZPoint MainVertex(0, 0, 0);
-  reco::Vertex TheMainVertex = vertexHandle.product()->front();
-
   if (vertexHandle.isValid()) {
     const reco::VertexCollection* vertices = vertexHandle.product();
-    if ((*vertices).at(0).isValid()) {
-      auto theMainVtx = (*vertices).at(0);
-      MainVertex.SetXYZ(theMainVtx.position().x(), theMainVtx.position().y(), theMainVtx.position().z());
-    }
+    theClosestVertex = this->findClosestVertex(aTransientVertex, vertices);
+  } else {
+    edm::LogWarning("DiMuonVertexMonitor") << "invalid vertex collection encountered Skipping event!";
+    return;
   }
 
+  reco::Vertex TheMainVertex;
+  if (!useClosestVertex_ || theClosestVertex == nullptr) {
+    TheMainVertex = *theClosestVertex;
+  } else {
+    TheMainVertex = vertexHandle.product()->front();
+  }
```

Comment:
Uhm, shouldn't it be the opposite?
```suggestion
  if (!useClosestVertex_ || theClosestVertex == nullptr) {
    TheMainVertex = vertexHandle.product()->front();
  } else {
    TheMainVertex = *theClosestVertex;
  }
```